classlib/src/main/java/org/teavm/classlib/java/util/TEnumMap.java

```diff
@@ -107,6 +97,10 @@ public V get(Object key) {
 
     @Override
     public V put(K key, V value) {
+        Class<?> cls = key.getClass();
+        if (cls != keyType && cls.getSuperclass() != keyType) {
```

Comment:
I agree that this is right, but things like these don't add much business value to TeaVM. What is your motivation to propose such changes?

Reply:
My motivation was to fix all found issues in Harmony tests.